components/service/experiments/src/main/java/mozilla/components/service/experiments/ExperimentsUpdater.kt

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.experiments
+
+import android.content.Context
+import android.support.annotation.VisibleForTesting
+import androidx.work.*
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.GlobalScope
+import mozilla.components.support.base.log.logger.Logger
+
+class SyncWorker(context: Context, workerParams: WorkerParameters) : androidx.work.Worker(context, workerParams) {
+    override fun doWork(): Result {
+        return if (Experiments.updater!!.updateExperiments()) {
```

Comment:
This bit is a blocker for me.
I really want to not call into the static interface, as this is terrible for testing and mutual dependencies.

But how can i call back one specific instance from a `Worker`?
- I can't seem to pass an instance value to it (i see only the notion of int/string/... "in data" through `WorkerParams`.
- Making this an `inner class` of `ExperimentsUpdater` would make it possible to access it's instance-specific variables. However, the `WorkManager` code can't instantiate this because the construct of an `inner class` is not publically (or module wide?) visible, leading to a runtime exception.

Reply:
>     * Making this an `inner class` of `ExperimentsUpdater` would make it possible to access it's instance-specific variables. However, the `WorkManager` code can't instantiate this because the construct of an `inner class` is not publically (or module wide?) visible, leading to a runtime exception.

Mh, this is interesting. I've been using the `inner class` approach [here](https://github.com/mozilla-mobile/android-components/blob/6e99fd867d018b16e24cfbb2bd8897a774f804bc/components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/MetricsPingScheduler.kt#L56) without particular problems. Is the case really different?

Reply:
Hm, ok... let me push that to compare.